enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsKernelExtensionFactoryIT.java

```diff
@@ -110,4 +104,59 @@ public void mustLoadMetricsExtensionWhenConfigured() throws Exception
             }
         }
     }
+
+    @Test
+    public void showReplanEvents() throws Exception
+    {
+        //do a simple query to populate cache
+        try ( Transaction tx = db.beginTx() )
+        {
+            db.execute( "match (n:Label {name: 'Pontus'}) return n.name");
+            tx.success();
+        }
+        //add some data, should make plan stale
+        addNodes( 10 );
+        //now we should have to plan again
+        try ( Transaction tx = db.beginTx() )
+        {
+            db.execute( "match (n:Label {name: 'Pontus'}) return n.name" );
+            tx.success();
+        }
+        //add more data just to make sure metrics are flushed
+        addNodes( 1000 );
+
+        //now we should have one replan event
+        try ( BufferedReader reader = new BufferedReader( new FileReader( outputFile ) ) )
+        {
+            String[] headers = reader.readLine().split( "," );
+            int replanColumn = Arrays.binarySearch( headers, CypherMetrics.REPLAN_EVENTS);
+            assertThat( replanColumn, is( not( -1 ) ) );
+
+            // Now we can verify that the number of committed transactions should never decrease.
+            int replanEvents = 0;
+            String line;
+            while ( (line = reader.readLine()) != null )
+            {
+                String[] fields = line.split( "," );
+                int newReplanEvents = Integer.parseInt( fields[replanColumn] );
+                assertThat( newReplanEvents, greaterThanOrEqualTo( replanEvents ) );
```

Comment:
Mostly because I copy pasted an old test. But it starts at 0 and at some point it is increased to one so it is not strictly equal. The more interesting check is beneath the loop where I check `replanEvents == 1`, but the other check at least asserts that we are not doing something like 
`0 + 1928 - 1930 + 3` (why now that would happen)